### PR TITLE
Refactor backward compatibility related code

### PR DESCRIPTION
**Pull request by Wetlize**
###### Motivation for this change
Just tidying up the codebase to ease maintenance and improve its understanding by the other users. See the commit messages for more details about the changes.

### CODE REVIEW

File: org-roam-compat.el [hunk 1]
```diff
@@ -33,7 +33,89 @@
 ;;; Code:
 ;;;; Library Requires
 
+;;; Backports
+(when (< emacs-major-version 27)
+  ;; Emacs 26 does not have FOLLOW-SYMLINKS in `directory-files-recursively'
+  (defun directory-files-recursively (dir regexp
```
Comment (jethrokuan) [inline org-roam-compat.el:39]: While harmless in this case, I don't think it's a good idea to shadow the function like this. i.e. we should leave the `org-roam-` prefix here.

**Comment by Wetlize** [comment]
Should be good to go now.

**Review by jethrokuan — APPROVED** [review]

File: org-roam-dailies.el [hunk 2]
```diff
@@ -327,25 +327,6 @@ negative, find note N days in the future."
 (define-key org-roam-dailies-map (kbd "v") #'org-roam-dailies-capture-date)
 (define-key org-roam-dailies-map (kbd ".") #'org-roam-dailies-find-directory)
 
-(define-obsolete-function-alias
```
Comment (d12frosted) [inline org-roam-dailies.el:330]: While I agree with `org-roam--directory-files-recursively` which is exactly what `compat` file are about, IMO obsoletisation of a function should stay in the module where function is defined.

Comment (Wetlize) [reply org-roam-dailies.el:330]: I kind of disagree, because it still promotes the usage of the obsolete functions and litters stuff. WDYT @jethrokuan?

Comment (jethrokuan) [reply org-roam-dailies.el:330]: I'm ambivalent: I think having all obsolete declarations in a compat file helps with cleanup later on, but separates the function from its "alias". This is generally not an issue, it's a set-and-forget kind of thing. There are arguments for both sides, I guess it's just important to stick to one style.

**Pull request merged**